### PR TITLE
chore: conservatively use log::info instead of println for some expressions

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

If you want to see info level logs, you can do `NOIR_LOG=info nargo compile` by default if that environment variable is not set, then it will only print out error level logs

## Problem\*

Resolves 

## Documentation\*

Check one:
- [ ] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[Exceptional Case]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: tooling/nargo_cli/src/main.rs [hunk 1]
```diff
@@ -12,10 +12,14 @@ mod cli;
 mod errors;
 
 use color_eyre::config::HookBuilder;
+use env_logger::{Builder, Env};
 
 const PANIC_MESSAGE: &str = "This is a bug. We may have already fixed this in newer versions of Nargo so try searching for similar issues at https://github.com/noir-lang/noir/issues/.\nIf there isn't an open issue for this bug, consider opening one at https://github.com/noir-lang/noir/issues/new?labels=bug&template=bug_report.yml";
 
 fn main() {
+    let env = Env::default().filter_or("NOIR_LOG", "error"); // Default to 'error' if NOIR is not set
```
Comment (kevaundray) [inline tooling/nargo_cli/src/main.rs:20]:
```suggestion
    let env = Env::default().filter_or("NOIR_LOG", "error"); // Default to 'error' if NOIR_LOG is not set
```

**Comment by kevaundray** [comment]
I'll open an issue to standardize the categories of logging we have -- some logging like printing out ACIR should be done using println, though for things like "Constraint system successfully built" its a bit trickier because its informational (we don't want it info level logs to be printed out by default), but we also want it to be printed out to tell the user that their program was built successfully

**Review by TomAFrench — APPROVED** [review]
I'm ok with this. I'm a little concerned about hiding some of the warnings related to interacting with backends by default but we can see how that goes to see if we get a resurgence of user questions.

**Review by jfecher — APPROVED** [review]
I'm alright with this for now as well. We can re-evaluate it later if needed

**Pull request merged**